src/com/xquery/xqueryClientLayer.cpp

```diff
@@ -21,108 +20,109 @@ extern "C" {
+  mSfd = -1;
+  mCommand = std::string("");
+  mHost = nullptr;
+  mPort = nullptr;
+  mUsr = nullptr;
+  mPsw = nullptr;
+  mDbName = nullptr;
 }
 
 CXqueryClientLayer::~CXqueryClientLayer() = default;
 
-bool CXqueryClientLayer::parseParameters(char *paLayerParameter) {
+bool CXqueryClientLayer::parseParameters(char *paLayerParameter){
   bool retVal = true;
-  host = strtok(paLayerParameter, scmParameterSeperator);
-  port = strtok(nullptr, scmParameterSeperator);
-  dbName = strtok(nullptr, scmParameterSeperator);
-  usr = strtok(nullptr, scmParameterSeperator);
-  psw = strtok(nullptr, scmParameterSeperator);
-  if (nullptr == host || nullptr == port || nullptr == usr || nullptr == psw || nullptr == dbName) {
+  mHost = strtok(paLayerParameter, scmParameterSeperator);
+  mPort = strtok(nullptr, scmParameterSeperator);
+  mDbName = strtok(nullptr, scmParameterSeperator);
+  mUsr = strtok(nullptr, scmParameterSeperator);
+  mPsw = strtok(nullptr, scmParameterSeperator);
+  if(nullptr == mHost || nullptr == mPort || nullptr == mUsr || nullptr == mPsw || nullptr == mDbName){
     retVal = false;
   }
   return retVal;
 }
 
-forte::com_infra::EComResponse CXqueryClientLayer::openConnection(char *paLayerParameter) {
+forte::com_infra::EComResponse CXqueryClientLayer::openConnection(char *paLayerParameter){
   EComResponse retVal = e_InitTerminated;
-  if (e_Client == mFb->getComServiceType() && mFb->getNumRD() == 1 && mFb->getNumSD() == 1 &&
-      parseParameters(paLayerParameter)) {
-    sfd = basex_connect(host, port);
-    if (sfd == -1) {
-      DEVLOG_ERROR("Can not connect to BaseX server.\n");
-    } else {
-      int rc = basex_authenticate(sfd, usr, psw);
+  if(e_Client == mFb->getComServiceType() && mFb->getNumRD() == 1 && mFb->getNumSD() == 1 && parseParameters(paLayerParameter)){
+    mSfd = basex_connect(mHost, mPort);
+    if (mSfd == -1) {
+      DEVLOG_ERROR("Can not connect to BaseX server at '%s:%s'\n", mHost, mPort);
+    }else{
+      int rc = basex_authenticate(mSfd, mUsr, mPsw);
       if (rc == -1) {
         DEVLOG_ERROR("Access to DB denied.\n");
-      } else {
-        mFb->QO() = true_BOOL;
+      }else{
+        mFb->QO() = CIEC_BOOL(true);
         DEVLOG_INFO("Connected to DB.\n");
         openDB();
         retVal = e_InitOk;
       }
     }
-  } else {
-    DEVLOG_ERROR("[xquery] supports only CLIENT_1 with ID = xquery[IP:port; dbName; usr; psw]\n");
+  }else{
+     DEVLOG_ERROR("[xquery] supports only CLIENT_1 with ID = xquery[IP:port; dbName; usr; psw]\n");
   }
   return retVal;
 }
 
-void CXqueryClientLayer::openDB() {
+void CXqueryClientLayer::openDB(){
   char *r = nullptr;
   char *i = nullptr;
-  command = new char[strlen(dbName) + 6];
-  strcpy(command, "open ");
-  strcat(command, dbName);
-  int ret = basex_execute(sfd, command, &r, &i);
-  if (ret == -1) { // general (i/o or the like) error
+  mCommand = std::string("open ") + std::string(mDbName);
```

Comment:
this can be better done using C++'s new string literal: `"open"s. In order to use this you need to add a `using namespace std::literals::string_literals;`